### PR TITLE
chore(examples): Convert `api-routes` example to TypeScript

### PR DESCRIPTION
**Pull request by maxproske**
Converted API Routes example over to TypeScript to match the Contribution guidelines.

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm lint`
- [X] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Let's consider https://github.com/vercel/next.js/pull/38091#discussion_r911320691 on this and all future PRs!

**Comment by maxproske** [comment]
Thanks @balazsorban44 - I made your requested change.

I thought `function` over `const` only applied to pages, but I'll consider components going forward.

File: examples/api-routes/pages/api/people/[id].ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { people } from '../../../data'
+import { Person } from '../../../interfaces'
+
+type Error = {
+  message: string
+}
```
Comment (balazsorban44) [inline examples/api-routes/pages/api/people/[id].ts:7]: This should probably be called something else to not clash with the native `Error` class.

**Comment by maxproske** [comment]
@balazsorban44 Thanks! Made your requested change, and ready for re-review!

Can I also get confirmation that you recommend `export default function` over split `const`/`export` for components, in addition to pages and API routes in examples?

**Review by balazsorban44 — APPROVED** [review]
Thanks!

As for https://github.com/vercel/next.js/pull/38083#issuecomment-1179169825, I think, for consistency's sake, we should use `function` over `const/let` uniformly, yes. `default` exports might not be necessary though (unless a page or API route), because a file could export multiple things, but these are stylistic details and a grey area what we should enforce vs. prefer.

So (in my opinion) use `function` whenever possible, but at the minimum for API Routes and pages. Rest should be at least consistent throughout the example. :+1:

**Comment by balazsorban44** [comment]
Opened https://github.com/vercel/next.js/pull/38511 to discuss the above and add it to the contributing guidelines

**Pull request merged**